core/common/src/main/java/alluxio/conf/InstancedConfiguration.java

```diff
@@ -454,9 +454,43 @@ private void checkTieredLocality() {
   }
 
   private class UnresolvablePropertyException extends Exception {
-
     public UnresolvablePropertyException(String msg) {
       super(msg);
     }
   }
+
+  /**
+   * @return a configuration builder
+   */
+  public static Builder newBuilder() {
+    return new Builder();
+  }
+
+  /**
+   * Class for building configuration.
+   */
+  public static class Builder {
+    private final AlluxioProperties mProperties = new AlluxioProperties();
```

Comment:
It effectively contains the defaults. `AlluxioProperties` falls back on the `PropertyKey` default value if it doesn't contain a value for a key:

```java
  @Nullable
  public String get(PropertyKey key) {
    if (mUserProps.containsKey(key)) {
      return mUserProps.get(key).orElse(null);
    }
    // In case key is not the reference to the original key
    return PropertyKey.fromString(key.toString()).getDefaultValue();
  }
```

Reply:
How about the site-properties? Will it skip those values? Is that desired? I thought config priority was something like: defaults -> site-properties -> cluster defaults -> runtime?